### ISSUE TITLE
Intermittent errors when doing s3 parquet query

### ISSUE THREAD

**Issue by calebmadrigal** [duckdb/duckdb#5190]
### What happens?

Most of the time, when I do an s3 parquet query (`con.execute(f"select * from parquet_scan('s3://{S3_BUCKET}/tasks/models/entity_status_task/model_raw_data.parquet/*.parquet') limit 100;")`) using my test script (included in "To Reproduce" section below), I get this error:

```
Forbidden
Traceback (most recent call last):
  File "test.py", line 15, in <module>
    out = con.execute(f"select * from parquet_scan('s3://{S3_BUCKET}/tasks/models/entity_status_task/model_raw_data.parquet/*.parquet') limit 100;")
duckdb.Error: Invalid Error: HTTP GET error on 'https://REDACTED.s3.amazonaws.com/?continuation-token=1maGEJ31m/5HgdHHOMGQX8HkuEdnAmp25tshbAsbbXGCV0HUEpqqtKLu85lMaEkKbS4owlFKDZnvDO9x9Y%2BBraegViGKNarK8onA9rIWGgstxAnkWjqwsMZLE7VTqQi9d8U61jXd8fv/a/Lz//iz3Lnw4ot2bxlQCdg2U72KV66Ll00YzlFfwNw%2BoVF3Zxgm1NOP9I9K%2B0K7wrrGGw2G7Ug%3D%3D&encoding-type=url&list-type=2&prefix=tasks%2Fmodels%2Fentity_status_task%2Fmodel_raw_data.parquet%2F' (HTTP 403)
```

But if I keep running it, it eventually just works. I've had it fail between 3 and 42 times before eventually working.

### To Reproduce

Here is the script I am using to test:

```
import os
import duckdb

AWS_ACCESS_KEY = os.environ.get('AWS_ACCESS_KEY_ID')
AWS_SECRET_KEY = os.environ.get('AWS_SECRET_ACCESS_KEY')
S3_BUCKET = os.environ.get('S3_BUCKET')

con = duckdb.connect()
con.execute(f"""INSTALL httpfs;
LOAD httpfs;
SET s3_region='us-east-1';
SET s3_access_key_id='{AWS_ACCESS_KEY}';
SET s3_secret_access_key='{AWS_SECRET_KEY}';""")

out = con.execute(f"select * from parquet_scan('s3://{S3_BUCKET}/tasks/models/entity_status_task/model_raw_data.parquet/*.parquet') limit 100;")
data = out.fetchall()

print(data[0])
```

### OS:

Amazon Linux 2

### DuckDB Version:

0.5.1

### DuckDB Client:

Python

### Full Name:

Caleb Madrigal

### Affiliation:

madconsulting.ai

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by calebmadrigal** [comment]
Given that this error seems non-deterministic, if there is some way for me to get additional verbose output, that could be helpful in debugging.

**Comment by hannes** [comment]
Can you maybe try to put some data in a public bucket? as it stands its hard to reproduce the issue with the `REDACTED` in there

**Comment by calebmadrigal** [comment]
@hannes Try this:

```
import os
import duckdb

S3_BUCKET = 'duckdb-test-bucket'
S3_PATH = f's3://{S3_BUCKET}/test2.parquet'

con = duckdb.connect()
con.execute(f"""INSTALL httpfs;
LOAD httpfs;
SET s3_region='us-east-1';""")

out = con.execute(f"select * from parquet_scan('{S3_PATH}/*.parquet') limit 100;")
data = out.fetchall()

print(data[0])
```

**Comment by calebmadrigal** [comment]
@hannes I'd be up for doing some screen sharing / pair programming on this if you'd like to see. It's very strange how it goes from return "403 Forbidden" errors to just working occasionally.

**Comment by samansmink** [comment]
@calebmadrigal this issue should be fixed by https://github.com/duckdb/duckdb/pull/5248

The intermittent behaviour is due to the continuation token (for paging listobjectv2 calls) sometimes containing symbols that were incorrectly encoded

**Comment by calebmadrigal** [comment]
@samansmink How can I test this out? Any idea when the next pip release will be?

**Comment by Alex-Monahan** [comment]
Next release is Monday!

**Comment by calebmadrigal** [comment]
Any idea when the binary distribution for `duckdb==0.6.1.dev8` will be available?

**Comment by calebmadrigal** [comment]
@Mause Awesome. I confirmed this issue is fixed in `0.6.0`.
